### DISCUSSION TITLE
To play all the tracks in the playlist with the same volume

### DISCUSSION THREAD

**Discussion by johnmapeson** [iina/iina#5293] [Q&A] [answered]
Guys, is it possible to play all the tracks in the playlist — if some of them are louder than other — to play with the same actual volume? In other words, if the playlist consists of 10 tracks and the 5th track is significally louder than other, I don't want to spill my seventeenth cup of coffee on the laptop.

**Comment by low-batt** [comment, marked as answer]
For audio/video and subtitle rendering IINA is using a library from the [mpv](https://mpv.io/) project. Many mpv features not directly exposed in IINA's UI are still available using advanced settings.

Thus when I am asked about a feature I will search the net to see if the mpv player supports it. As the mpv player supports many platforms there are many people using it and you are likely to find people discussing the feature. I found mpv issue https://github.com/mpv-player/mpv/issues/2883, "mpv Audio Normalization: play all files at equal volume", which seems applicable. I believe the best solution is the one given in [this comment](https://github.com/mpv-player/mpv/issues/2883#issuecomment-189957200):
> The simplest, most effective and most supported solution is to scan your local files in advance (e.g. with mp3gain, flacgain, vorbisgain, etc.) and tag their RG information into the files.

See that mpv issue for full details and some other suggestions. That comment is referring to using [ReplayGain](https://en.wikipedia.org/wiki/ReplayGain). See that Wikipedia article for details on this standard.

IINA support for ReplayGain has already been requested in issue #4915. Support is present in IINA 1.3.5, there just isn't a UI to make it easy to use. Currently to use ReplayGain you have to add mpv options on the `Advanced` tab of IINA's settings.

A UI to make it easy to use ReplayGain has been implemented in PR #4932. See the screenshot in that PR for what the new settings look like. This enhancement is in the current development version of IINA and will be in the next official release of IINA. That PR also shows how I ran the [rsgain](https://github.com/complexlogic/rsgain) tool over some audio files to add ReplayGain tags to the files.

Using ReplayGain seems to be the standard way of normalizing volume across audio tracks.
